### PR TITLE
feat: add support for separate GitHub app credentials

### PR DESCRIPTION
**Pull request by dlactin**
This PR adds support for separate GitHub app credentials using the pre-existing GitHub App functions

### CODE REVIEW

**Review by jannfis — COMMENTED** [review]
Change LGTM so far, thank you!

I have just one request regarding error handling.

File: pkg/argocd/gitcreds.go [hunk 1]
```diff
@@ -66,14 +67,25 @@ func getCredsFromSecret(wbc *WriteBackConfig, credentialsSecret string, kubeClie
...
+		if githubAppID, ok = credentials["githubAppID"]; ok {
+			if githubAppInstallationID, ok = credentials["githubAppInstallationID"]; !ok {
+				return nil, fmt.Errorf("invalid secret %s: does not contain field githubAppInstallationID", credentialsSecret)
+			}
+			if githubAppPrivateKey, ok = credentials["githubAppPrivateKey"]; !ok {
+				return nil, fmt.Errorf("invalid secret %s: does not contain field githubAppPrivateKey", credentialsSecret)
+			}
+			// converting byte array to string and ultimately int64 for NewGitHubAppCreds
+			intGithubAppID, _ := strconv.ParseInt(string(githubAppID), 10, 64)
+			intGithubAppInstallationID, _ := strconv.ParseInt(string(githubAppInstallationID), 10, 64)
```
[the remark this answers is not in the document]
Comment (dlactin) [reply pkg/argocd/gitcreds.go:80]: Ah yes, meant to add that after testing. I'll add it now!

File: pkg/argocd/gitcreds.go [hunk 2]
```diff
@@ -66,14 +67,31 @@ func getCredsFromSecret(wbc *WriteBackConfig, credentialsSecret string, kubeClie
...
+			if githubAppInstallationID, ok = credentials["githubAppInstallationID"]; !ok {
+				return nil, fmt.Errorf("invalid secret %s: does not contain field githubAppInstallationID", credentialsSecret)
+			}
+			if githubAppPrivateKey, ok = credentials["githubAppPrivateKey"]; !ok {
+				return nil, fmt.Errorf("invalid secret %s: does not contain field githubAppPrivateKey", credentialsSecret)
+			}
+			// converting byte array to string and ultimately int64 for NewGitHubAppCreds
+			intGithubAppID, err := strconv.ParseInt(string(githubAppID), 10, 64)
+			if err != nil {
+				return nil, err
```
Comment (jannfis) [inline pkg/argocd/gitcreds.go:81]:
```suggestion
				return nil, fmt.Errorf("invalid value in field githubAppID: %w", err) 
```
just a tiny improvement to help with troubleshooting

File: pkg/argocd/gitcreds.go [hunk 3]
```diff
@@ -66,14 +67,31 @@ func getCredsFromSecret(wbc *WriteBackConfig, credentialsSecret string, kubeClie
...
+			}
+			intGithubAppInstallationID, _ := strconv.ParseInt(string(githubAppInstallationID), 10, 64)
+			if err != nil {
+				return nil, err
```
Comment (jannfis) [inline pkg/argocd/gitcreds.go:85]:
```suggestion
				return nil, fmt.Errorf("invalid value in field githubAppInstallationID: %w", err) 

```
just a tiny improvement to help with troubleshooting

**Review by jannfis — COMMENTED** [review]
Small suggestion for returning the error, so that the error message may be more helpful to the user.

**Review by jannfis — APPROVED** [review]
LGTM!

Thanks a lot for this PR, @dlactin. Appreciate it much!

**Pull request merged**